### ISSUE TITLE
Content type is missed in content manager sometimes

### ISSUE THREAD

**Issue by henrych4** [strapi/strapi#2355]
**Informations**
- **Node.js version**: 10.10.0
- **npm version**: 6.4.1
- **Strapi version**: 3.0.0-alpha.14.5
- **Database**: mongodb
- **Operating system**: mac OS

**What is the current behavior?**
Sometimes, content types are missing in content manager.

**Steps to reproduce the problem**
Try to create a few content types and check if all of them appear in content manager.
This may be hard to reproduce since it is related to the order of models loaded.

**What is the expected behavior?**
All content types should appear in content manager.

**Suggested solutions**
After adding some logs to the code, I found that the problem comes from `getModels` method:
https://github.com/strapi/strapi/blob/eeca356060e6754c14be4baf89e317f7015264ed/packages/strapi-plugin-content-manager/admin/src/containers/SettingsPage/index.js#L40-L58

In my case, one of the content types is iterated before `plugins` and is accumulated. After that the next content type is `plugins`, so `getModels` is called recursively. After returning from recursive calls, previous accumulated value is replaced by the return value, leading to missing content type in content manager.

Concatenating the retuned result to previous accumulated value solves the problem.

**Comment by henrych4** [comment]
After double checked the content manager, I found that there were an extra field `File` which was generated from plugin: upload. Clicking on it causes error so the fix may not be suitable.

And then I have some thought: The content manager should show those content types on the LeftMenu. So why don't we get the data from layout (`this.props.schema.layout`) instead of from models (`this.props.schema.models`) ? In this way, we don't even need the hard coded filtering:
```
models.filter(obj => obj.name !== 'permission' && obj.name !== 'role')
```

How do you guys think? I can work on it if we all agree to fix in such way.

**Comment by soupette** [comment]
@henrych4 the issue has been fixed is #2325 I changed the recursive. I've looked into the code from #2356 and I think that your recursive is way better!

Regarding 
> The content manager should show those content types on the LeftMenu.
Those content types are editable through the `roles & permissions` link on the left menu and I'm not sure @Aurelsicoko wants them to appear in it.

> why don't we get the data from layout (this.props.schema.layout) instead of from models

That's something we can do but you'll need some logic to generate the correct urls for the `SettingPage` since I use the url to update the correct setting in the content type.

**Comment by henrych4** [comment]
I have tried the code in pull request #2325 but still, one of content types is missed on content manager. 

> That's something we can do but you'll need some logic to generate the correct urls for the `SettingPage` since I use the url to update the correct setting in the content type.

You are right. It seems that url cannot be determined from layout only. How about using the layout content to determine whether a model should be shown in content manager?
ie.
```suggestion
models.filter(obj => !!this.props.schema.layout[obj.name])
```
